src/routes/InstancesPage/InstancesToolbarSearchFilter.js

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ToolbarToggleGroup,
+  ToolbarGroup,
+  ToolbarFilter,
+  SelectVariant,
+  SelectOption,
+  Select,
+  ToolbarItem,
+  InputGroup,
+  TextInput,
+  Button,
+} from '@patternfly/react-core';
+import { FilterIcon, SearchIcon } from '@patternfly/react-icons';
+import SelectSingle from '../../components/SelectSingle';
+
+function InstancesToolbarSearchFilter({ filters, setFilters }) {
+  const [selectedFilter, setSelectedFilter] = useState('Name');
+  // local state for input values
+  const [inputName, setInputName] = useState('');
+  const [inputOwner, setInputOwner] = useState('');
+  // local state for Select isExpanded values
+  // @TODO: We can refactor the SelectSingle component to be more reusable for the usecase in this component as well. Then we don't need to keep this state here.
+  const [isRegionExpanded, setIsRegionExpanded] = useState(false);
+  const [isStatusExpanded, setIsStatusExpanded] = useState(false);
+
+  // TODO: Extract into separate utils file to be reused in other cases
+  function onDeleteChip(type = '', id = '') {
+    if (type) {
+      setFilters((prevFilters) => {
+        const newFilters = { ...prevFilters };
+        newFilters[type.toLowerCase()] = newFilters[type.toLowerCase()].filter(
+          (s) => s !== id
+        );
+        return newFilters;
+      });
+    } else {
+      setFilters({
+        region: [],
+      });
+    }
+  }
+
+  // TODO: Extract into separate utils file to be reused in other cases
+  function onDeleteChipGroup(type) {
+    setFilters((prevFilters) => {
+      const newFilters = { ...prevFilters };
+      newFilters[type.toLowerCase()] = [];
+      return newFilters;
+    });
+  }
+
+  // TODO: Extract into separate utils file to be reused in other cases
+  function onSelect(type, event, selection) {
+    const checked = event.target.checked;
+    setFilters((prevFilters) => {
+      const prevSelections = prevFilters[type] || [];
```

Comment:
Would it be beneficial to force the `toLowerCase()` here too? It looks like the only current usage is via the two functions below, but would be safer/more consistent with the `onDelete` util functions.

Reply:
Ah, so I don't think this one really needs the `toLowerCase()`. Here we explicitly pass the type. So in `onRegionSelect` and `onStatusSelect` we pass the type as the first argument. For the `onDeleteChip` and `onDeleteChipGroup` we do need to do it because we capitalize the `categoryName` because that's what's displayed when we render the chips in those grouping